src/core/InstrumentPlayHandle.cpp

```diff
@@ -37,5 +40,41 @@ InstrumentPlayHandle::InstrumentPlayHandle( Instrument * instrument, InstrumentT
 	setAudioPort( instrumentTrack->audioPort() );
 }
 
+void InstrumentPlayHandle::play( sampleFrame * _working_buffer )
+{
+	InstrumentTrack * instrumentTrack = m_instrument->instrumentTrack();
+
+	// ensure that all our nph's have been processed first
+	ConstNotePlayHandleList nphv = NotePlayHandle::nphsOfInstrumentTrack(instrumentTrack, true );
```

Comment:
This line is a bit long. I would recommend we use ``auto`` here.
```suggestion
	auto nphv = NotePlayHandle::nphsOfInstrumentTrack(instrumentTrack, true);
```